src/wallet/mod.rs

```diff
@@ -559,6 +565,47 @@ impl Wallet {
 		Ok(txid)
 	}
 
+	pub(crate) fn select_confirmed_utxos(
```

Comment:
Would be cool to be able to reuse this for `impl CoinSelectionSource` while dropping `impl WalletSource` going forward. What do we think the necessary changes would be that would allow for this? Probably just making `claim_id` optional?

Reply:
cc: @wpaulino 

Hmm... are you saying have LDK Node's `Wallet` implement `CoinSelectionSource` directly? Would we implement UTXO locking there? Or are you assuming BDK's wallet will eventually implement this?

To re-use this code, yeah we can extract the `Utxo` from each `FundingTxInput` to use in `CoinSelection`. Though we need to expose that or have a conversion function. Currently, `FundingTxInput::utxo` is `pub(super)`.

Reply:
> Hmm... are you saying have LDK Node's `Wallet` implement `CoinSelectionSource` directly? 

Yes, it might be a bit more efficient than the detour over `WalletSource`, and would allow us to reuse most code, IIUC.

> Would we implement UTXO locking there? Or are you assuming BDK's wallet will eventually implement this?

We could, although I still prefer to wait until BDK ships it soon hopefully. Might be debatable if we need to implement it manually in the meantime. However, the UTXO locking discussion seems orthogonal to whether or not to use `CoinSelectionSource`, no?